x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/predicate/operator/arithmetic/DateTimeArithmeticOperation.java

```diff
@@ -50,16 +54,49 @@ protected TypeResolution resolveType() {
         DataType rightType = right().dataType();
         // date math is only possible if one argument is a DATETIME and the other a (foldable) TemporalValue
         if (isDateTimeOrTemporal(leftType) || isDateTimeOrTemporal(rightType)) {
-            if (argumentOfType(DataTypes::isDateTime) == null || argumentOfType(EsqlDataTypes::isTemporalAmount) == null) {
-                return new TypeResolution(
-                    format(null, "[{}] has arguments with incompatible types [{}] and [{}]", symbol(), leftType, rightType)
-                );
+            if (argumentOfType(DataTypes::isDateTime) != null && argumentOfType(EsqlDataTypes::isTemporalAmount) != null) {
+                return TypeResolution.TYPE_RESOLVED;
             }
-            return TypeResolution.TYPE_RESOLVED;
+            if (EsqlDataTypes.isTemporalAmount(leftType) && EsqlDataTypes.isTemporalAmount(rightType)) {
+                return TypeResolution.TYPE_RESOLVED;
+            }
```

Comment:
These two conditions look redundant - isn't argumentOfType already checking the TemporalAmount?

Reply:
The first condition checks whether there is one datetime and one duration argument (where duration does not include datetime) in any order. The second checks whether both arguments are durations, so the two conditions are mutually exclusive.

I'll rewrite this in a way that is hopefully clearer!